sisl/viz/plotly/plotutils.py

```diff
@@ -672,14 +672,18 @@ def run_multiple(func, *args, argsList = None, kwargsList = None, messageFn = No
     """
 
     #Prepare the arguments to be passed to the initSinglePlot function
+    # TODO could itertools.zip_longest perhaps be used instead?
+    # it sure smells like it?
```

Comment:
Hmm no, because what `zip_longest` do is to pad all shorter iterables with a fill value.

Here if `toZip` is `[2, 3, [1,4], [3,5,6]]`, what we want to end up zipping is `zip([2,2], [3,3], [1,4], [3,5,6])`.

But this part in general is poorly written I think. I have to check, but I don't think the `isinstance` check is necessary, the only things that are already prepared to be zipped are `argsList` and `kwargsList`. The instance check looks very prone to bugs.

Reply:
ok.